InfiniteRecharge/src/main/java/frc/robot/Robot.java

```diff
@@ -71,19 +71,19 @@ public void robotInit() {
         visionSubsystem = new VisionSubsystem(config);
         subsystems.add(visionSubsystem);
 
+        if (config.enableShooterSubsystem) {
+            shooterSubsystem = new ShooterSubsystem(config, visionSubsystem);
+            subsystems.add(shooterSubsystem);
+        }
+
         if (config.enableDriveSubsystem) {
             navigationSubsystem = new NavigationSubsystem(config, visionSubsystem);
-            driveSubsystem = new DriveSubsystem(config, navigationSubsystem, oi);
+            driveSubsystem = new DriveSubsystem(config, navigationSubsystem, shooterSubsystem, oi);

```

Comment:
I don't think it's right to make the shooterSubsystem a dependency on the driveSubsystem. The driveSubsystem should use the vision subsystem, not the shooter. All the driveSubsystem is getting from the shooter is the tx from the vision.

Reply:
my reasoning was that we want the filtered value that the shooter subsystem has, but I just realized I'd need some (option to) filter with yaw in the same way that the shooter filters using the shooter position. will give it some more thought and see what I come up with